scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -383,14 +383,27 @@ object ScioContext {
     } else {
       val args = Args(appArgs)
       if (appArgs.nonEmpty) {
+        val argString = args.toString("", ", ", "")
+        val sanitizedArgString =
+          if (argString.length > appArgStringMaxLength) {
+            log.warn("Truncating long app arguments")
+            argString.substring(0, appArgStringMaxLength)
```

Comment:
Minor but I'd add a suffix like `"..."` to indicate that it'd been truncated.

Reply:
yeah +1 on this!